src/Command/Handler/CreateArticle.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Command\Handler;
+
+use App\Command\Command;
+use App\Command\CommandResult;
+use App\Command\CreateArticleResult;
+use App\Entity\Article;
+use App\Repository\ArticleRepository;
+use Ausi\SlugGenerator\SlugGenerator;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class CreateArticle implements CommandHandler
+{
+    /**
+     * @var ObjectManager
+     */
+    private $manager;
+
+    public function __construct(
+        ObjectManager $manager
+    ) {
+        $this->manager = $manager;
+    }
+
+    public function handle(Command $command): CommandResult
+    {
+        // @TODO this should rather be injected
+        $generator = new SlugGenerator();
+
+        $article = Article::write(
+            $command->title,
+            $command->content,
+            $generator->generate($command->title)
+        );
+
+        $this->manager->persist($article);
+        $this->manager->flush();
```

Comment:
It’s very cool to use the entity manager’s methods directly in your handler, as it makes the code clear and explicit, but in case you get tired of this, you can still use a middleware (if the chosen command bus allows it, and it should) to handle auto-flushing, transaction, etc.

Reply:
Yep for now I do it that way because its straight forward and minimal + the command bus I use is very minimal too (it just resolves the command hadler based on the command class and then executes the handler over the command)